File: typescript/src/getTicker.ts
import axios from 'axios';
import * as dotenv from 'dotenv';

dotenv.config();

async function getTicker(symbol: string): Promise<void> {
  try {
    const response = await axios.get(`${process.env.BASE_URL}/ticker`, {
      params: { symbol },
      headers: {
        'X-API-Key': process.env.API_KEY,
      },
    });

    console.log('Ticker data:', response.data);
  } catch (error) {
    console.error('Error fetching ticker:', error);
  }
}

// Example usage
getTicker('BTCUSD');

File: python/get_ticker.py
import os
import requests
from dotenv import load_dotenv

load_dotenv()

def get_ticker(symbol: str):
    """Fetch ticker data for a given symbol."""
    url = f"{os.getenv('BASE_URL')}/ticker"
    headers = {
        'X-API-Key': os.getenv('API_KEY')
    }
    params = {'symbol': symbol}

    try:
        response = requests.get(url, headers=headers, params=params)
        response.raise_for_status()
        print('Ticker data:', response.json())
    except requests.exceptions.RequestException as e:
        print(f'Error fetching ticker: {e}')

if __name__ == '__main__':
    get_ticker('BTCUSD')
